src/components/Workspace.jsx

```diff
@@ -133,9 +138,88 @@ export default class Workspace extends React.Component {
     );
   }
 
+  _renderHiddenLanguages() {
+    const {
+      currentProject,
+      hiddenLanguages,
+      onComponentToggle,
+    } = this.props;
+    return (
+      <React.Fragment>
+        {hiddenLanguages.map(({language}) =>
+          (
+            <CollapsedComponent
+              component={`editor.${language}`}
+              key={language}
+              projectKey={currentProject.projectKey}
+              text={t(`languages.${language}`)}
+              onComponentUnhide={onComponentToggle}
+            />
+          ))}
+      </React.Fragment>
+    );
+  }
+
+  _renderHiddenLeftColumnComponents() {
+    const {
+      hiddenLanguages,
+    } = this.props;
+
+    if (
+      hiddenLanguages.length === 0 ||
+      hiddenLanguages.length === LANGUAGES.length
+    ) {
+      return null;
+    }
+
+    return this._renderHiddenLanguages();
+  }
+
+  _renderHiddenRightColumnComponents() {
+    const {
+      currentProject,
+      onComponentToggle,
+    } = this.props;
+    const rightColumnComponents = ['console'];
+    return rightColumnComponents.
+      filter(component => (
+        includes(currentProject.hiddenUIComponents, component)
+      )).
+      map((component) => {
+        switch (component) {
+          case 'console':
+            return (
+              <CollapsedComponent
+                component="console"
+                isRightJustified={false}
+                key="console"
+                projectKey={currentProject.projectKey}
+                text={t('workspace.components.console')}
+                onComponentUnhide={onComponentToggle}
+              />
+            );
+          default:
+            return null;
+        }
+      });
+  }
+
+  _renderFullyMinimizedColumns() {
+    if (!this._shouldRenderLeftColumn()) {
+      return this._renderHiddenLanguages();
+    }
+    return null;
+  }
+
+  _shouldRenderLeftColumn() {
+    return !(this.props.hiddenLanguages.length === LANGUAGES.length);
```

Comment:
More idiomatic to use `!==` (or in this case `<` would work) rather than negating the whole expression